### PR TITLE
Cleanup docs examples

### PR DESCRIPTION
**Pull request by provinzkraut**
This is part of the documentation overhaul project, aimed at starting the process to solve #343 in particular. It does a few things

- Add a small CLI utility to find and extract code examples in the docs, with the added option to scaffold tests for them
- Extract examples from and add tests for:
  - Path params
  - Query params
  - Header and cookie params
  - Layered params
  - Returning responses
  - Response headers
  - Response cookies
  - Streaming responses
- Rewrite the "Query params" section (because it was hard to make succinct, self-contained examples for it)

# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

### CODE REVIEW

**Comment by provinzkraut** [comment]
Opening this for review now, since I don't want it to get too big. I'm gonna follow up with a few more PRs, so this is going to be a sequential process.

File: examples/responses/response_cookies_4.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+def after_request_handler(response: Response) -> Response:
+    response.set_cookie(**Cookie(key="Random-Cookie", value=str(randint(1, 100))).dict)
+    return response
+
+
+router = Router(
+    path="/router-path",
+    route_handlers=[retrieve_resource],
+    after_request=after_request_handler,
+    response_cookies=[
```
Comment (jtraub) [inline examples/responses/response_cookies_4.py:31]: @provinzkraut could you please check. It looks like a real bug.

Adding `response_cookies` param to the router makes Starlite to emit `set-cookie` header twice. In this case second header has empty value.

File: tools/doc_examples.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    parser_check.set_defaults(func=check_command)
+    parser_extract.set_defaults(func=extract_command)
+
+    args = parser.parse_args()
+    args.func(args)
+
+
+if __name__ == "__main__":
+    cli()
```
Comment (Goldziher) [inline tools/doc_examples.py:172]: can you document this cli ?

**Review by Goldziher — APPROVED** [review]

File: tools/doc_examples.py (hunk 2 above)
Comment (provinzkraut) [reply tools/doc_examples.py:172]: WDYM? Or rather, where do you think it should be documented?

File: examples/responses/response_cookies_4.py (hunk 1 above)
Comment (provinzkraut) [reply examples/responses/response_cookies_4.py:31]: I `xfaile`ed this test for now.

**Pull request merged**